components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -146,15 +146,26 @@ open class GleanInternalAPI {
 
     /**
      * Collect and assemble the ping. Asynchronously submits the assembled
-     * payload to the designated server using [httpPingUploader].
+     * payload to the designated server using [httpPingUploader] but only
+     * if metrics are enabled.
      */
     internal fun sendPing(store: String, docType: String) {
+        // Do not send ping if metrics disabled
+        if (!getMetricsEnabled()) {
+            return
```

Comment:
nit: let's print a log line here, trace or debug level, mentioning what we're trying to send and that we're bailing out because upload is disabled.